odl/solvers/functional/default_functionals.py

```diff
@@ -312,6 +304,60 @@ def __repr__(self):
         return '{}({!r})'.format(self.__class__.__name__, self.domain)
 
 
+# TODO make l1 and l2 norms inherit from this.
+class LpNorm(Functional):
+
+    """The functional corresponding to the Lp-norm.
+
+    The Lp-norm, ``||f||_p``,  is defined as
+
+    .. math::
+        \\left(\\int |f(x)|^p dx \\right)^{1/p}
```

Comment:
Write something about the special cases `inf, -inf, 0`. In particular the case `p=0` can have an additional sentence mentioning that it isn't a norm in the strict sense (like the others for `p<1`), and that it is strongly discretization-dependent (counting nonzeros keeps growing for growing number of discretization points). Perhaps a general formulation would be that the norms with `p < 1` have been added only for completeness.